module/applications/item/item-sheet.mjs

```diff
@@ -109,4 +118,39 @@ export default class ItemSheetEd extends ItemSheet {
     const effect = this.item.effects.get( li.dataset.itemId );
     return effect.sheet?.render( true );
   }
+
+  /* ----------------------------------------------------------------------- */
+  /*                Auto calculation for equipments weight                   */
+  /* ----------------------------------------------------------------------- */
+
+  /**
+   * Handle autorecalculation of physical items for actors, based on the namegiver modifier for weight.
+   */
+  async _onWeightCalculation( ) {
```

Comment:
Here's a more version that might be more streamlined:
```javascript
async _onWeightCalculation() {
    const item = this.item;
    const namegiver = item.parent.namegiver;
    if ( item.isOwned && !item.system.weight.weightCalculated && namegiver ) {
      const updateData = {
        "name": `${item.name} (${namegiver.name})`,
        "system.weight.value": namegiver.system.weightMultiplier * item.system.weight.value,
        "system.weight.weightCalculated": true,
        "system.weight.weightMultiplier": namegiver.system.weightMultiplier,
      }
      await item.update( updateData );
      this.render( true );
    } else if ( item.system.weight.weightCalculated ) {
      ui.notifications.warn( game.i18n.localize("X.cantUpdateItemWeight" ) );
    }
  }
```
If a weight is zero, we can still update it. It's just zero afterwards. And since this function not only calcs weight (should think about renaming it), but basically 'tailors' it to the namgiver, some users might wanna use it just for that, even when they don't use weights at all.
Also, we need to call `update` on the item, otherwise the changes are not persisted.

Reply:
ok moved and updated

Reply:
Streamlined in commit 74f3682a5f5384e2ae4986733dec0a073f2fde18.